src/ferc_xbrl_extractor/datapackage.py

```diff
@@ -182,9 +201,8 @@ def _lowercase_words(name: str) -> str:
     return name
 
 
-def _clean_table_names(name: str) -> Optional[str]:
-    """
-    Function to clean table names.
+def _clean_table_names(name: str) -> str | None:
```

Comment:
Maybe this is all clean enough now that it doesn't matter, but if you find yourself chasing down all manner of different special characters, it turns out there are some string normalization functions built into the standard Python libraries, and Python validators that mutate the value during validation seem like they might be a good fit for taking an existing table name and outputting strict snake case.  I think there's a `SnakeCase` model in `pudl.metadata.classes` but don't know if it does what you need.

Reply:
Hmm I just read up on the pydantic `constr` [here](https://pydantic-docs.helpmanual.io/usage/types/#arguments-to-constr) used for `SnakeCase`, and it seems like it can't quite handle everything I need. From my understanding it can do things like convert to lowercase and strip whitespace, but the regex used for `SnakeCase` is only used for validation, and doesn't actually do any normalization. I'm not 100% sure I'm reading the docs correctly though

Reply:
Ah okay, that sounds right that our SnakeCase is just validating, and not mutating.

The standard library unicode normalization function I recently came across was [unicodedata.normalize](https://docs.python.org/3/library/unicodedata.html#module-unicodedata).

Sounds like there's nothing off the shelf that will do what's necessary here, so looks good to me.